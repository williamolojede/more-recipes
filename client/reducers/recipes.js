/**
 * Reducer for All Recipe
 */

import { RECIEVE_TOP_RATED_RECIPE } from '../actions/types';

const recipes = (state = [], action) => {
  switch (action.type) {
    case RECIEVE_TOP_RATED_RECIPE:
      return action.recipes;
    default:
      return state;
  }
};


export default recipes;
