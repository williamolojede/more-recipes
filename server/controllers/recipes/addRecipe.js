import { Recipe } from '../../models/index';

const addRecipe = (req, res, next) => {
  const { name, description, img_url, ingredients, instructions } = req.body;
  Recipe.create({
    name,
    description,
    img_url,
    ingredients,
    instructions,
    owner: req.userID
  })
    .then(recipe => res.status(201).json({ recipe, message: 'success' }))
    .catch((error) => {
      const err = new Error(error);
      err.status = 500;
      next(err);
    });
};

export default addRecipe;
