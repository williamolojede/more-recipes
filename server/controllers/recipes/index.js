import createRecipe from './addRecipe';
import getAllRecipe from './getAllRecipe';
import updateRecipe from './updateRecipe';
import deleteRecipe from './deleteRecipe';

const recipesController = {
  getAllRecipe,
  createRecipe,
  updateRecipe,
  deleteRecipe
};


export default recipesController;
