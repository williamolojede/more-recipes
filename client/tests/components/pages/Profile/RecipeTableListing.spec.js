import React from 'react';
import RecipeTableListing from '../../../../components/pages/Profile/RecipeTableListing';
import recipes, { favoriteRecipes } from '../../../../__mocks__/recipes.mock';

function setup(type, asOwner, recipesData) {
  const props = {
    type,
    asOwner,
    recipes: recipesData,
    removeRecipe: jest.fn()
  };
  const wrapper = shallow(<RecipeTableListing {...props} />);

  return {
    wrapper
  };
}


describe('RecipeTableListing', () => {
  describe('type: Favorites', () => {
    const type = 'favorites';
    it('should render with right amount of elements if no recipes', () => {
      const { wrapper } = setup(type, true, []);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render with right amount of elements if no recipes and asOwner: false', () => {
      const { wrapper } = setup(type, false, []);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render list of recipes without actions if not owner', () => {
      const { wrapper } = setup(type, false, favoriteRecipes());
      expect(wrapper.find('th').length).toBe(1);
      expect(wrapper.find('RecipeTableListingRow').length).toBe(wrapper.instance().state.limit);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render list of recipes with actions if owner', () => {
      const { wrapper } = setup(type, true, favoriteRecipes());
      expect(wrapper.find('th').length).toBe(2);
      expect(wrapper.find('RecipeTableListingRow').length).toBe(wrapper.instance().state.limit);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('type: Personal-Recipes', () => {
    const type = 'personal-recipes';
    it('should render with right amount of elements if no recipes', () => {
      const { wrapper } = setup(type, true, []);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render with right amount of elements if no recipes and asOwner: false', () => {
      const { wrapper } = setup(type, false, []);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render list of recipes without actions if not owner', () => {
      const { wrapper } = setup(type, false, recipes);
      expect(wrapper.find('th').length).toBe(1);
      expect(wrapper.find('RecipeTableListingRow').length).toBe(wrapper.instance().state.limit);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render list of recipes with actions if owner', () => {
      const { wrapper } = setup(type, true, recipes);
      expect(wrapper.find('th').length).toBe(2);
      expect(wrapper.find('RecipeTableListingRow').length).toBe(wrapper.instance().state.limit);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('getRecipesForPage', () => {
    it('should do nothing if called to get recipes for the currentPage', () => {
      const { wrapper } = setup('favorites', true, []);
      expect(wrapper.instance().getRecipesForPage(1)).toBe(undefined);
      expect(wrapper.instance().state.currentPage)
        .toBe(1);
    });

    it('should do nothing if called to get recipes for pages outside available pages', () => {
      const { wrapper } = setup('personal-recipes', true, recipes);
      expect(wrapper.instance().getRecipesForPage(-1)).toBe(undefined);
      expect(wrapper.instance().getRecipesForPage(100)).toBe(undefined);
      expect(wrapper.instance().state.currentPage)
        .toBe(1);
    });

    it('should update currentpage state when called within range', () => {
      const { wrapper } = setup('personal-recipes', true, recipes);
      wrapper.instance().getRecipesForPage(2);
      expect(wrapper.instance().state.currentPage).toBe(2);
      // this will break depending on the length of `recipes` and `...state.limit`
      expect(wrapper.instance().state.recipes.length)
        .toBe(recipes.length % wrapper.instance().state.limit);
    });
  });
});
