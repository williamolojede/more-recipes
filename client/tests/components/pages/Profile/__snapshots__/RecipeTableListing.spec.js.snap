// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeTableListing type: Favorites should render list of recipes with actions if owner 1`] = `
<table
  className="my-recipe__manage--table"
>
  <thead>
    <tr>
      <th
        colSpan={null}
      >
        Name
      </th>
      <th>
        Actions
      </th>
    </tr>
  </thead>
  <tbody>
    <RecipeTableListingRow
      asOwner={true}
      index={0}
      key="2"
      recipe={
        Object {
          "User": Object {
            "fullname": "example user",
            "id": 2,
            "username": null,
          },
          "createdAt": "2017-10-23T08:08:47.566Z",
          "description": "It is simply better than ghanian jollof",
          "downVoteCount": 1,
          "favoriteCount": 3,
          "id": 2,
          "img_url": "http://www.africanbites.com/wp-content/uploads/2014/05/IMG_9677-2-1-150x150.jpg",
          "ingredients": Array [
            "rice",
            "canola oil",
          ],
          "instructions": Array [
            "Break up the clumpy rice before starting.",
            "Garnish with chopped scallion and serve",
          ],
          "name": "Jollof Rice",
          "owner": 2,
          "upVoteCount": 3,
          "updatedAt": "2018-01-16T10:02:13.814Z",
          "viewCount": 6,
        }
      }
      removeRecipe={[Function]}
      type="favorites"
    />
    <RecipeTableListingRow
      asOwner={true}
      index={1}
      key="5"
      recipe={
        Object {
          "User": Object {
            "fullname": "Email Mail",
            "id": 11,
            "username": null,
          },
          "createdAt": "2017-11-09T09:08:15.109Z",
          "description": "This is one of my 'go to' recipes and it takes only about 15 minutes from start to finish. The only prep work really is mashing the garlic. The sauce is great on any pasta, but I prefer angel hair or spaghetti.",
          "downVoteCount": 1,
          "favoriteCount": 1,
          "id": 5,
          "img_url": "http://images.media-allrecipes.com/userphotos/720x405/4478567.jpg",
          "ingredients": Array [
            "2 tablespoons olive oil",
            "3 cloves garlic",
            "2 tablespoons capers",
            "1 tablespoon anchovy paste",
            "1 teaspoon red pepper flakes",
            "1 (24 ounce) jar tomato pasta sauce",
            "1/4 cup water",
            "1 cup good-quality crisp white wine",
            "2 (6.5 ounce) cans chopped clams, drained with juice reserved",
            "salt and pepper to taste",
            "1 pound dry spaghetti",
            "1/4 cup freshly grated Parmesan cheese",
            "Chopped fresh basil leaves or parsley",
          ],
          "instructions": Array [
            "Pour olive oil into cold skillet. Add garlic, capers, anchovy paste, and red pepper flakes. Place over medium heat and cook and stir until the oil is infused with the flavors of the mixture, about 5 minutes, taking care not to brown the garlic. Add tomato",
            "Bring a large pot of lightly salted water to a boil. Cook spaghetti in the boiling water, stirring occasionally until almost al dente, or about 1 minute less than directed on package. Drain. Return pasta to the pot off heat.",
            "Stir clams into the sauce. Pour sauce over pasta. Stir; cover the pot to allow it to finish cooking off heat and to allow the pasta will absorb some of the sauce, about 3 minutes. Stir in Parmesan cheese. Top with chopped fresh basil and/or parsley.",
          ],
          "name": "Chef John's Spaghetti with Red Clam Sauce dfbefadb nifek fedckltefdkletfdmlefdleff eifdanlkref",
          "owner": 11,
          "upVoteCount": 1,
          "updatedAt": "2018-01-10T10:54:26.519Z",
          "viewCount": 2,
        }
      }
      removeRecipe={[Function]}
      type="favorites"
    />
    <RecipeTableListingRow
      asOwner={true}
      index={2}
      key="169"
      recipe={
        Object {
          "User": Object {
            "fullname": "William Olojede",
            "id": 29,
            "username": null,
          },
          "createdAt": "2018-01-13T00:09:15.646Z",
          "description": "This is the recipe for a very delicious dish",
          "downVoteCount": 0,
          "favoriteCount": 0,
          "id": 169,
          "img_url": "https://imgurlfordish.com/verydelicousdish.jpg",
          "ingredients": Array [
            "ingredient 1",
            "ingredient 2",
          ],
          "instructions": Array [
            "This is the first step in cooking this delicious dish",
            "This is the next step",
          ],
          "name": "some random very delicious dish",
          "owner": 29,
          "upVoteCount": 1,
          "updatedAt": "2018-01-13T10:50:00.734Z",
          "viewCount": 2,
        }
      }
      removeRecipe={[Function]}
      type="favorites"
    />
    <RecipeTableListingRow
      asOwner={true}
      index={3}
      key="25"
      recipe={
        Object {
          "User": Object {
            "fullname": "Email Mail",
            "id": 11,
            "username": null,
          },
          "createdAt": "2017-11-11T09:07:11.059Z",
          "description": "This is one of my 'go to' recipes ",
          "downVoteCount": 0,
          "favoriteCount": 1,
          "id": 25,
          "img_url": "http://images.media-allrecipes.com/userphotos/720x405/4478567.jpg",
          "ingredients": Array [
            "2 tablespoons olive oil",
            "3 cloves garlic",
            "2 tablespoons capers",
            "1 tablespoon anchovy paste",
            "1 teaspoon red pepper flakes",
            "1 (24 ounce) jar tomato pasta sauce",
            "1/4 cup water",
            "1 cup good-quality crisp white wine",
            "2 (6.5 ounce) cans chopped clams, drained with juice reserved",
            "salt and pepper to taste",
            "1 pound dry spaghetti",
            "1/4 cup freshly grated Parmesan cheese",
            "Chopped fresh basil leaves or parsley",
          ],
          "instructions": Array [
            "Pour olive oil into cold skillet. Add garlic, capers, anchovy paste, and red pepper flakes. Place over medium heat and cook and stir until the oil is infused with the flavors of the mixture, about 5 minutes, taking care not to brown the garlic. Add tomato",
            "Bring a large pot of lightly salted water to a boil. Cook spaghetti in the boiling water, stirring occasionally until almost al dente, or about 1 minute less than directed on package. Drain. Return pasta to the pot off heat.",
            "Stir clams into the sauce. Pour sauce over pasta. Stir; cover the pot to allow it to finish cooking off heat and to allow the pasta will absorb some of the sauce, about 3 minutes. Stir in Parmesan cheese. Top with chopped fresh basil and/or parsley.",
          ],
          "name": "Chef John's Spaghetti with Red Clam Sauce dfbefadb nifek fedckltefdkletfdmlefdleff eifdanlkref",
          "owner": 11,
          "upVoteCount": 1,
          "updatedAt": "2017-11-26T15:54:12.531Z",
          "viewCount": 2,
        }
      }
      removeRecipe={[Function]}
      type="favorites"
    />
  </tbody>
  <tfoot>
    <tr>
      <td
        colSpan="2"
      >
        <Pagination
          currentPage={1}
          getRecipesForPage={[Function]}
          pages={
            Array [
              1,
              2,
            ]
          }
        />
      </td>
    </tr>
  </tfoot>
</table>
`;

exports[`RecipeTableListing type: Favorites should render list of recipes without actions if not owner 1`] = `
<table
  className="my-recipe__manage--table"
>
  <thead>
    <tr>
      <th
        colSpan="2"
      >
        Name
      </th>
    </tr>
  </thead>
  <tbody>
    <RecipeTableListingRow
      asOwner={false}
      index={0}
      key="2"
      recipe={
        Object {
          "User": Object {
            "fullname": "example user",
            "id": 2,
            "username": null,
          },
          "createdAt": "2017-10-23T08:08:47.566Z",
          "description": "It is simply better than ghanian jollof",
          "downVoteCount": 1,
          "favoriteCount": 3,
          "id": 2,
          "img_url": "http://www.africanbites.com/wp-content/uploads/2014/05/IMG_9677-2-1-150x150.jpg",
          "ingredients": Array [
            "rice",
            "canola oil",
          ],
          "instructions": Array [
            "Break up the clumpy rice before starting.",
            "Garnish with chopped scallion and serve",
          ],
          "name": "Jollof Rice",
          "owner": 2,
          "upVoteCount": 3,
          "updatedAt": "2018-01-16T10:02:13.814Z",
          "viewCount": 6,
        }
      }
      removeRecipe={[Function]}
      type="favorites"
    />
    <RecipeTableListingRow
      asOwner={false}
      index={1}
      key="5"
      recipe={
        Object {
          "User": Object {
            "fullname": "Email Mail",
            "id": 11,
            "username": null,
          },
          "createdAt": "2017-11-09T09:08:15.109Z",
          "description": "This is one of my 'go to' recipes and it takes only about 15 minutes from start to finish. The only prep work really is mashing the garlic. The sauce is great on any pasta, but I prefer angel hair or spaghetti.",
          "downVoteCount": 1,
          "favoriteCount": 1,
          "id": 5,
          "img_url": "http://images.media-allrecipes.com/userphotos/720x405/4478567.jpg",
          "ingredients": Array [
            "2 tablespoons olive oil",
            "3 cloves garlic",
            "2 tablespoons capers",
            "1 tablespoon anchovy paste",
            "1 teaspoon red pepper flakes",
            "1 (24 ounce) jar tomato pasta sauce",
            "1/4 cup water",
            "1 cup good-quality crisp white wine",
            "2 (6.5 ounce) cans chopped clams, drained with juice reserved",
            "salt and pepper to taste",
            "1 pound dry spaghetti",
            "1/4 cup freshly grated Parmesan cheese",
            "Chopped fresh basil leaves or parsley",
          ],
          "instructions": Array [
            "Pour olive oil into cold skillet. Add garlic, capers, anchovy paste, and red pepper flakes. Place over medium heat and cook and stir until the oil is infused with the flavors of the mixture, about 5 minutes, taking care not to brown the garlic. Add tomato",
            "Bring a large pot of lightly salted water to a boil. Cook spaghetti in the boiling water, stirring occasionally until almost al dente, or about 1 minute less than directed on package. Drain. Return pasta to the pot off heat.",
            "Stir clams into the sauce. Pour sauce over pasta. Stir; cover the pot to allow it to finish cooking off heat and to allow the pasta will absorb some of the sauce, about 3 minutes. Stir in Parmesan cheese. Top with chopped fresh basil and/or parsley.",
          ],
          "name": "Chef John's Spaghetti with Red Clam Sauce dfbefadb nifek fedckltefdkletfdmlefdleff eifdanlkref",
          "owner": 11,
          "upVoteCount": 1,
          "updatedAt": "2018-01-10T10:54:26.519Z",
          "viewCount": 2,
        }
      }
      removeRecipe={[Function]}
      type="favorites"
    />
    <RecipeTableListingRow
      asOwner={false}
      index={2}
      key="169"
      recipe={
        Object {
          "User": Object {
            "fullname": "William Olojede",
            "id": 29,
            "username": null,
          },
          "createdAt": "2018-01-13T00:09:15.646Z",
          "description": "This is the recipe for a very delicious dish",
          "downVoteCount": 0,
          "favoriteCount": 0,
          "id": 169,
          "img_url": "https://imgurlfordish.com/verydelicousdish.jpg",
          "ingredients": Array [
            "ingredient 1",
            "ingredient 2",
          ],
          "instructions": Array [
            "This is the first step in cooking this delicious dish",
            "This is the next step",
          ],
          "name": "some random very delicious dish",
          "owner": 29,
          "upVoteCount": 1,
          "updatedAt": "2018-01-13T10:50:00.734Z",
          "viewCount": 2,
        }
      }
      removeRecipe={[Function]}
      type="favorites"
    />
    <RecipeTableListingRow
      asOwner={false}
      index={3}
      key="25"
      recipe={
        Object {
          "User": Object {
            "fullname": "Email Mail",
            "id": 11,
            "username": null,
          },
          "createdAt": "2017-11-11T09:07:11.059Z",
          "description": "This is one of my 'go to' recipes ",
          "downVoteCount": 0,
          "favoriteCount": 1,
          "id": 25,
          "img_url": "http://images.media-allrecipes.com/userphotos/720x405/4478567.jpg",
          "ingredients": Array [
            "2 tablespoons olive oil",
            "3 cloves garlic",
            "2 tablespoons capers",
            "1 tablespoon anchovy paste",
            "1 teaspoon red pepper flakes",
            "1 (24 ounce) jar tomato pasta sauce",
            "1/4 cup water",
            "1 cup good-quality crisp white wine",
            "2 (6.5 ounce) cans chopped clams, drained with juice reserved",
            "salt and pepper to taste",
            "1 pound dry spaghetti",
            "1/4 cup freshly grated Parmesan cheese",
            "Chopped fresh basil leaves or parsley",
          ],
          "instructions": Array [
            "Pour olive oil into cold skillet. Add garlic, capers, anchovy paste, and red pepper flakes. Place over medium heat and cook and stir until the oil is infused with the flavors of the mixture, about 5 minutes, taking care not to brown the garlic. Add tomato",
            "Bring a large pot of lightly salted water to a boil. Cook spaghetti in the boiling water, stirring occasionally until almost al dente, or about 1 minute less than directed on package. Drain. Return pasta to the pot off heat.",
            "Stir clams into the sauce. Pour sauce over pasta. Stir; cover the pot to allow it to finish cooking off heat and to allow the pasta will absorb some of the sauce, about 3 minutes. Stir in Parmesan cheese. Top with chopped fresh basil and/or parsley.",
          ],
          "name": "Chef John's Spaghetti with Red Clam Sauce dfbefadb nifek fedckltefdkletfdmlefdleff eifdanlkref",
          "owner": 11,
          "upVoteCount": 1,
          "updatedAt": "2017-11-26T15:54:12.531Z",
          "viewCount": 2,
        }
      }
      removeRecipe={[Function]}
      type="favorites"
    />
  </tbody>
  <tfoot>
    <tr>
      <td
        colSpan="2"
      >
        <Pagination
          currentPage={1}
          getRecipesForPage={[Function]}
          pages={
            Array [
              1,
              2,
            ]
          }
        />
      </td>
    </tr>
  </tfoot>
</table>
`;

exports[`RecipeTableListing type: Favorites should render with right amount of elements if no recipes 1`] = `
<div
  className="no-favorites"
>
  <span>
    Welcome to your favorites!
  </span>
  <span>
    Tap 
    <i
      className="mdi mdi-heart"
    />
     to save any recipe you like,
  </span>
  <span>
    and it'll show up here.
  </span>
</div>
`;

exports[`RecipeTableListing type: Favorites should render with right amount of elements if no recipes and asOwner: false 1`] = `
<div
  className="no-favorites"
>
  <span>
    This user as no favorite recipes yet...
  </span>
</div>
`;

exports[`RecipeTableListing type: Personal-Recipes should render list of recipes with actions if owner 1`] = `
<table
  className="my-recipe__manage--table"
>
  <thead>
    <tr>
      <th
        colSpan={null}
      >
        Name
      </th>
      <th>
        Actions
      </th>
    </tr>
  </thead>
  <tbody>
    <RecipeTableListingRow
      asOwner={true}
      index={0}
      key="2"
      recipe={
        Object {
          "User": Object {
            "fullname": "example user",
            "id": 2,
            "username": null,
          },
          "createdAt": "2017-10-23T08:08:47.566Z",
          "description": "It is simply better than ghanian jollof",
          "downVoteCount": 1,
          "favoriteCount": 3,
          "id": 2,
          "img_url": "http://www.africanbites.com/wp-content/uploads/2014/05/IMG_9677-2-1-150x150.jpg",
          "ingredients": Array [
            "rice",
            "canola oil",
          ],
          "instructions": Array [
            "Break up the clumpy rice before starting.",
            "Garnish with chopped scallion and serve",
          ],
          "name": "Jollof Rice",
          "owner": 2,
          "upVoteCount": 3,
          "updatedAt": "2018-01-16T10:02:13.814Z",
          "viewCount": 6,
        }
      }
      removeRecipe={[Function]}
      type="personal-recipes"
    />
    <RecipeTableListingRow
      asOwner={true}
      index={1}
      key="5"
      recipe={
        Object {
          "User": Object {
            "fullname": "Email Mail",
            "id": 11,
            "username": null,
          },
          "createdAt": "2017-11-09T09:08:15.109Z",
          "description": "This is one of my 'go to' recipes and it takes only about 15 minutes from start to finish. The only prep work really is mashing the garlic. The sauce is great on any pasta, but I prefer angel hair or spaghetti.",
          "downVoteCount": 1,
          "favoriteCount": 1,
          "id": 5,
          "img_url": "http://images.media-allrecipes.com/userphotos/720x405/4478567.jpg",
          "ingredients": Array [
            "2 tablespoons olive oil",
            "3 cloves garlic",
            "2 tablespoons capers",
            "1 tablespoon anchovy paste",
            "1 teaspoon red pepper flakes",
            "1 (24 ounce) jar tomato pasta sauce",
            "1/4 cup water",
            "1 cup good-quality crisp white wine",
            "2 (6.5 ounce) cans chopped clams, drained with juice reserved",
            "salt and pepper to taste",
            "1 pound dry spaghetti",
            "1/4 cup freshly grated Parmesan cheese",
            "Chopped fresh basil leaves or parsley",
          ],
          "instructions": Array [
            "Pour olive oil into cold skillet. Add garlic, capers, anchovy paste, and red pepper flakes. Place over medium heat and cook and stir until the oil is infused with the flavors of the mixture, about 5 minutes, taking care not to brown the garlic. Add tomato",
            "Bring a large pot of lightly salted water to a boil. Cook spaghetti in the boiling water, stirring occasionally until almost al dente, or about 1 minute less than directed on package. Drain. Return pasta to the pot off heat.",
            "Stir clams into the sauce. Pour sauce over pasta. Stir; cover the pot to allow it to finish cooking off heat and to allow the pasta will absorb some of the sauce, about 3 minutes. Stir in Parmesan cheese. Top with chopped fresh basil and/or parsley.",
          ],
          "name": "Chef John's Spaghetti with Red Clam Sauce dfbefadb nifek fedckltefdkletfdmlefdleff eifdanlkref",
          "owner": 11,
          "upVoteCount": 1,
          "updatedAt": "2018-01-10T10:54:26.519Z",
          "viewCount": 2,
        }
      }
      removeRecipe={[Function]}
      type="personal-recipes"
    />
    <RecipeTableListingRow
      asOwner={true}
      index={2}
      key="169"
      recipe={
        Object {
          "User": Object {
            "fullname": "William Olojede",
            "id": 29,
            "username": null,
          },
          "createdAt": "2018-01-13T00:09:15.646Z",
          "description": "This is the recipe for a very delicious dish",
          "downVoteCount": 0,
          "favoriteCount": 0,
          "id": 169,
          "img_url": "https://imgurlfordish.com/verydelicousdish.jpg",
          "ingredients": Array [
            "ingredient 1",
            "ingredient 2",
          ],
          "instructions": Array [
            "This is the first step in cooking this delicious dish",
            "This is the next step",
          ],
          "name": "some random very delicious dish",
          "owner": 29,
          "upVoteCount": 1,
          "updatedAt": "2018-01-13T10:50:00.734Z",
          "viewCount": 2,
        }
      }
      removeRecipe={[Function]}
      type="personal-recipes"
    />
    <RecipeTableListingRow
      asOwner={true}
      index={3}
      key="25"
      recipe={
        Object {
          "User": Object {
            "fullname": "Email Mail",
            "id": 11,
            "username": null,
          },
          "createdAt": "2017-11-11T09:07:11.059Z",
          "description": "This is one of my 'go to' recipes ",
          "downVoteCount": 0,
          "favoriteCount": 1,
          "id": 25,
          "img_url": "http://images.media-allrecipes.com/userphotos/720x405/4478567.jpg",
          "ingredients": Array [
            "2 tablespoons olive oil",
            "3 cloves garlic",
            "2 tablespoons capers",
            "1 tablespoon anchovy paste",
            "1 teaspoon red pepper flakes",
            "1 (24 ounce) jar tomato pasta sauce",
            "1/4 cup water",
            "1 cup good-quality crisp white wine",
            "2 (6.5 ounce) cans chopped clams, drained with juice reserved",
            "salt and pepper to taste",
            "1 pound dry spaghetti",
            "1/4 cup freshly grated Parmesan cheese",
            "Chopped fresh basil leaves or parsley",
          ],
          "instructions": Array [
            "Pour olive oil into cold skillet. Add garlic, capers, anchovy paste, and red pepper flakes. Place over medium heat and cook and stir until the oil is infused with the flavors of the mixture, about 5 minutes, taking care not to brown the garlic. Add tomato",
            "Bring a large pot of lightly salted water to a boil. Cook spaghetti in the boiling water, stirring occasionally until almost al dente, or about 1 minute less than directed on package. Drain. Return pasta to the pot off heat.",
            "Stir clams into the sauce. Pour sauce over pasta. Stir; cover the pot to allow it to finish cooking off heat and to allow the pasta will absorb some of the sauce, about 3 minutes. Stir in Parmesan cheese. Top with chopped fresh basil and/or parsley.",
          ],
          "name": "Chef John's Spaghetti with Red Clam Sauce dfbefadb nifek fedckltefdkletfdmlefdleff eifdanlkref",
          "owner": 11,
          "upVoteCount": 1,
          "updatedAt": "2017-11-26T15:54:12.531Z",
          "viewCount": 2,
        }
      }
      removeRecipe={[Function]}
      type="personal-recipes"
    />
  </tbody>
  <tfoot>
    <tr>
      <td
        colSpan="2"
      >
        <Pagination
          currentPage={1}
          getRecipesForPage={[Function]}
          pages={
            Array [
              1,
              2,
            ]
          }
        />
      </td>
    </tr>
  </tfoot>
</table>
`;

exports[`RecipeTableListing type: Personal-Recipes should render list of recipes without actions if not owner 1`] = `
<table
  className="my-recipe__manage--table"
>
  <thead>
    <tr>
      <th
        colSpan="2"
      >
        Name
      </th>
    </tr>
  </thead>
  <tbody>
    <RecipeTableListingRow
      asOwner={false}
      index={0}
      key="2"
      recipe={
        Object {
          "User": Object {
            "fullname": "example user",
            "id": 2,
            "username": null,
          },
          "createdAt": "2017-10-23T08:08:47.566Z",
          "description": "It is simply better than ghanian jollof",
          "downVoteCount": 1,
          "favoriteCount": 3,
          "id": 2,
          "img_url": "http://www.africanbites.com/wp-content/uploads/2014/05/IMG_9677-2-1-150x150.jpg",
          "ingredients": Array [
            "rice",
            "canola oil",
          ],
          "instructions": Array [
            "Break up the clumpy rice before starting.",
            "Garnish with chopped scallion and serve",
          ],
          "name": "Jollof Rice",
          "owner": 2,
          "upVoteCount": 3,
          "updatedAt": "2018-01-16T10:02:13.814Z",
          "viewCount": 6,
        }
      }
      removeRecipe={[Function]}
      type="personal-recipes"
    />
    <RecipeTableListingRow
      asOwner={false}
      index={1}
      key="5"
      recipe={
        Object {
          "User": Object {
            "fullname": "Email Mail",
            "id": 11,
            "username": null,
          },
          "createdAt": "2017-11-09T09:08:15.109Z",
          "description": "This is one of my 'go to' recipes and it takes only about 15 minutes from start to finish. The only prep work really is mashing the garlic. The sauce is great on any pasta, but I prefer angel hair or spaghetti.",
          "downVoteCount": 1,
          "favoriteCount": 1,
          "id": 5,
          "img_url": "http://images.media-allrecipes.com/userphotos/720x405/4478567.jpg",
          "ingredients": Array [
            "2 tablespoons olive oil",
            "3 cloves garlic",
            "2 tablespoons capers",
            "1 tablespoon anchovy paste",
            "1 teaspoon red pepper flakes",
            "1 (24 ounce) jar tomato pasta sauce",
            "1/4 cup water",
            "1 cup good-quality crisp white wine",
            "2 (6.5 ounce) cans chopped clams, drained with juice reserved",
            "salt and pepper to taste",
            "1 pound dry spaghetti",
            "1/4 cup freshly grated Parmesan cheese",
            "Chopped fresh basil leaves or parsley",
          ],
          "instructions": Array [
            "Pour olive oil into cold skillet. Add garlic, capers, anchovy paste, and red pepper flakes. Place over medium heat and cook and stir until the oil is infused with the flavors of the mixture, about 5 minutes, taking care not to brown the garlic. Add tomato",
            "Bring a large pot of lightly salted water to a boil. Cook spaghetti in the boiling water, stirring occasionally until almost al dente, or about 1 minute less than directed on package. Drain. Return pasta to the pot off heat.",
            "Stir clams into the sauce. Pour sauce over pasta. Stir; cover the pot to allow it to finish cooking off heat and to allow the pasta will absorb some of the sauce, about 3 minutes. Stir in Parmesan cheese. Top with chopped fresh basil and/or parsley.",
          ],
          "name": "Chef John's Spaghetti with Red Clam Sauce dfbefadb nifek fedckltefdkletfdmlefdleff eifdanlkref",
          "owner": 11,
          "upVoteCount": 1,
          "updatedAt": "2018-01-10T10:54:26.519Z",
          "viewCount": 2,
        }
      }
      removeRecipe={[Function]}
      type="personal-recipes"
    />
    <RecipeTableListingRow
      asOwner={false}
      index={2}
      key="169"
      recipe={
        Object {
          "User": Object {
            "fullname": "William Olojede",
            "id": 29,
            "username": null,
          },
          "createdAt": "2018-01-13T00:09:15.646Z",
          "description": "This is the recipe for a very delicious dish",
          "downVoteCount": 0,
          "favoriteCount": 0,
          "id": 169,
          "img_url": "https://imgurlfordish.com/verydelicousdish.jpg",
          "ingredients": Array [
            "ingredient 1",
            "ingredient 2",
          ],
          "instructions": Array [
            "This is the first step in cooking this delicious dish",
            "This is the next step",
          ],
          "name": "some random very delicious dish",
          "owner": 29,
          "upVoteCount": 1,
          "updatedAt": "2018-01-13T10:50:00.734Z",
          "viewCount": 2,
        }
      }
      removeRecipe={[Function]}
      type="personal-recipes"
    />
    <RecipeTableListingRow
      asOwner={false}
      index={3}
      key="25"
      recipe={
        Object {
          "User": Object {
            "fullname": "Email Mail",
            "id": 11,
            "username": null,
          },
          "createdAt": "2017-11-11T09:07:11.059Z",
          "description": "This is one of my 'go to' recipes ",
          "downVoteCount": 0,
          "favoriteCount": 1,
          "id": 25,
          "img_url": "http://images.media-allrecipes.com/userphotos/720x405/4478567.jpg",
          "ingredients": Array [
            "2 tablespoons olive oil",
            "3 cloves garlic",
            "2 tablespoons capers",
            "1 tablespoon anchovy paste",
            "1 teaspoon red pepper flakes",
            "1 (24 ounce) jar tomato pasta sauce",
            "1/4 cup water",
            "1 cup good-quality crisp white wine",
            "2 (6.5 ounce) cans chopped clams, drained with juice reserved",
            "salt and pepper to taste",
            "1 pound dry spaghetti",
            "1/4 cup freshly grated Parmesan cheese",
            "Chopped fresh basil leaves or parsley",
          ],
          "instructions": Array [
            "Pour olive oil into cold skillet. Add garlic, capers, anchovy paste, and red pepper flakes. Place over medium heat and cook and stir until the oil is infused with the flavors of the mixture, about 5 minutes, taking care not to brown the garlic. Add tomato",
            "Bring a large pot of lightly salted water to a boil. Cook spaghetti in the boiling water, stirring occasionally until almost al dente, or about 1 minute less than directed on package. Drain. Return pasta to the pot off heat.",
            "Stir clams into the sauce. Pour sauce over pasta. Stir; cover the pot to allow it to finish cooking off heat and to allow the pasta will absorb some of the sauce, about 3 minutes. Stir in Parmesan cheese. Top with chopped fresh basil and/or parsley.",
          ],
          "name": "Chef John's Spaghetti with Red Clam Sauce dfbefadb nifek fedckltefdkletfdmlefdleff eifdanlkref",
          "owner": 11,
          "upVoteCount": 1,
          "updatedAt": "2017-11-26T15:54:12.531Z",
          "viewCount": 2,
        }
      }
      removeRecipe={[Function]}
      type="personal-recipes"
    />
  </tbody>
  <tfoot>
    <tr>
      <td
        colSpan="2"
      >
        <Pagination
          currentPage={1}
          getRecipesForPage={[Function]}
          pages={
            Array [
              1,
              2,
            ]
          }
        />
      </td>
    </tr>
  </tfoot>
</table>
`;

exports[`RecipeTableListing type: Personal-Recipes should render with right amount of elements if no recipes 1`] = `
<div
  className="no-favorites"
>
  <span>
    You don't have any personal recipes yet.
  </span>
  <span>
    Share your own culinary creations! Add a recipe
  </span>
</div>
`;

exports[`RecipeTableListing type: Personal-Recipes should render with right amount of elements if no recipes and asOwner: false 1`] = `
<div
  className="no-favorites"
>
  <span>
    This user as no personal recipes yet...
  </span>
</div>
`;
